javaScriptAJAX-programming/codeChallenge/roomWidget.js

```diff
@@ -0,0 +1,26 @@
+let getRooms =  new XMLHttpRequest();
+
+getRooms.onreadystatechange = function () {
+  if (getRooms.readyState === 4) {
+    if(getRooms.status === 200) {
+      const rooms = JSON.parse(getRooms.responseText);
+      let getRoomStatusHTML = '<ul class = "rooms">';
+
+      for(let i = 0; i < rooms.length; i += 1) {
+        if(rooms[i].available) {
+          getRoomStatusHTML += '<li class = "empty">';
+        } else {
+          getRoomStatusHTML += '<li class = "full">';
+        }
+        getRoomStatusHTML += rooms[i].room;
+        getRoomStatusHTML += '</li>';
+      }
+      getRoomStatusHTML += '</ul>';
+
+      document.getElementById('roomList').innerHTML = getRoomStatusHTML;
+    }
```

Comment:
Good job on this part, @nicoRDL! However, there are a couple of things I'd like to point out:

- The spacing after keywords such as `if`, `for`, and `else` is inconsistent. For better readability, there should be a space after those keywords. Consider automating code formatting by installing a **Prettier** extension or a similar tool.

- The spaces after the class attribute are unnecessary. There should be no space after the `class` attribute and after the equal sign.

- Furthermore, this logic could be shortened by using **string literals**, but that would be a nice-to-have improvement.

These suggestions will help enhance the code's clarity and maintainability.

Reply:
Noted on all change requests and suggestions. 4d6c1e6f87851ef55c8c6f2e3a6b8d50777bac7c

Formatter applied to file. Additional code refactoring applied. For your re-review @lycoangelo

Reply:
@nicoRDL, there's still no space after the keyword `if`. It should be `if (condition) {}`

Reply:
Hi @lycoangelo. I think the PR did not update on your end. Latest commit (https://github.com/nicoRDL/dotTheory-JS/commit/4d6c1e6f87851ef55c8c6f2e3a6b8d50777bac7c) already includes the updated spacing between conditionals and class attributes. Requesting re-review.

Current File:
<img width="1434" alt="Screenshot 2023-09-04 at 2 12 29 PM" src="https://github.com/nicoRDL/dotTheory-JS/assets/100089185/bd245d91-a0c5-4f32-8755-d096be84b3a7">

Reply:
Thank you for your prompt response, @nicoRDL! I realized my mistake as I was actually examining the `javaScriptAJAX-programming/checkResponseStatus/sampleAJAX.js file`.